cli/fabric.c

```diff
@@ -37,9 +37,19 @@
 #include <errno.h>
 #include <ctype.h>
 
+static const char * const cmd_desc_topo_info = "Show topology information";
+static const char * const cmd_desc_gfms_bind = "Bind the EP (function) to the specified host";
+static const char * const cmd_desc_gfms_unbind = "Unbind the EP (function) from the specified host";
+static const char * const cmd_desc_gfms_dump = "Dump the GFMS database";
+static const char * const cmd_desc_route = "Show routing information";
+static const char * const cmd_desc_port_control = "Control a port";
+static const char * const cmd_desc_portcfg_show = "Display a port's configuration information";
+static const char * const cmd_desc_portcfg_set = "Configure a port";
+static const char * const cmd_desc_gfms_events = "Display GFMS event information";
+
```

Comment:
Nak -- please keep the descriptions with the commands. Then people will be more likely to see them if they need to update them.

Reply:
Note I'll be changing these from variables to #defines since this compiles fine in Windows but not Linux. Apparently certain versions of certain compilers have a limitation where they don't recognize such variable declarations as constant (even though they're declared with const), so the compiler errors out when assigning them to struct elements that expect const.

Do you still think it's better to keep the definitions right above the associated functions? Typically #define's are grouped together at the top of the file, before any function definitions, as this keeps things more organized. It seems like that's the switchtec-user convention too.

Reply:
What? That's mad. I know GCC can do this fine. #define's are not really appropriate here.

This works fine for me on any version of GCC/clang, I try: 
https://godbolt.org/z/9iUEAD

You might have an unnecessary second const there that doesn't really make sense in that context; but it should still compile.

And yes, still keep it by the function.

Reply:
It fails when used like this: https://godbolt.org/z/y62_5o
That's how we're using it here. GCC 7.3 fails, and GCC 8.1+ succeeds.

Reply:
Gross. I'd just leave it the way it was then instead of dealing with #define mess. 

Technically the desc in the function is supposed to be more detailed than the short one-line description in the list. They were never supposed to be the same.

Reply:
They were all the same the way things were before these changes, except for minor things like a missing "the" or a spelling mistake. And some of the descriptions were entirely incorrect in one of the instances but not the other as they were pasted from other commands. If I revert these back then I should at least fix any errors, but that would basically duplicate each description. The duplication is probably what led to certain instances having a correction but others not, which is why duplication should typically be avoided.

If the descriptions in the function were mostly different than the commands[] descriptions as originally intended then I agree we should leave these as they were, but since they're basically all duplicated I think the #defines would be the safer option.

Reply:
Ok, well do what you feel is best.